test/ngAnimate/integrationSpec.js

```diff
@@ -756,5 +756,38 @@ describe('ngAnimate integration tests', function() {
         expect(child.attr('style')).toContain('50px');
       });
     });
+
+
+    it('should execute the enter animation on a <form> with ngIf that has a <input> of type="email" that is required', function() {
```

Comment:
This test replicates the actual case from the original issue, the test in animateSpec.js is the "condensed" version. In practice it's pretty rare, because you actually need all these exact directives. The root cause lies in the way the email input sets / unsets the errors / error classes. Because for some reason, email and url inputs set ctrl.$$parserType even though they don't have a parser. That adds some unnessecary class additions / removals which trigger this bug. I actually meant to open a PR for fixing this, but forgot.